### DISCUSSION TITLE
FastAPIError:  Invalid args for response field! When using dependencies

### DISCUSSION THREAD

**Discussion by jramiromolina** [fastapi/fastapi#10828] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.post(
    "/",
    response_model=ContentResponse,
    dependencies=[AuthenticatedUser],

)
async def create_content(
        *,
        session: Session = ActiveSession,
        request: Request,
        content: ContentIncoming,
):
    # set the ownsership of the content to the current user
    db_content = Content.from_orm(content)
    user: User = get_current_user(request=request)
    db_content.user_id = user.id
    session.add(db_content)
    session.commit()
    session.refresh(db_content)
    return db_content
```

### Description

I started recently to build and app using FastApi. And I get a FastApiError exception when using dependencies in a route. 
If I remove the dependency everything works but when I add it I get the exception about the response type.

The whole exception is 

ImportError while loading conftest '/home/konradzuse/PycharmProjects/application-service/tests/conftest.py'.
tests/conftest.py:11: in <module>
    from app_service import app, settings, db  # noqa
app_service/__init__.py:1: in <module>
    from .app import app
app_service/app.py:9: in <module>
    from .routes import main_router
app_service/routes/__init__.py:3: in <module>
    from .content import router as content_router
app_service/routes/content.py:35: in <module>
    @router.post(
.venv/lib64/python3.11/site-packages/fastapi/routing.py:921: in decorator
    self.add_api_route(
.venv/lib64/python3.11/site-packages/fastapi/routing.py:860: in add_api_route
    route = route_class(
.venv/lib64/python3.11/site-packages/fastapi/routing.py:494: in __init__
    get_parameterless_sub_dependant(depends=depends, path=self.path_format),
.venv/lib64/python3.11/site-packages/fastapi/dependencies/utils.py:125: in get_parameterless_sub_dependant
    return get_sub_dependant(depends=depends, dependency=depends.dependency, path=path)
.venv/lib64/python3.11/site-packages/fastapi/dependencies/utils.py:148: in get_sub_dependant
    sub_dependant = get_dependant(
.venv/lib64/python3.11/site-packages/fastapi/dependencies/utils.py:269: in get_dependant
    sub_dependant = get_param_sub_dependant(
.venv/lib64/python3.11/site-packages/fastapi/dependencies/utils.py:112: in get_param_sub_dependant
    return get_sub_dependant(
.venv/lib64/python3.11/site-packages/fastapi/dependencies/utils.py:148: in get_sub_dependant
    sub_dependant = get_dependant(
.venv/lib64/python3.11/site-packages/fastapi/dependencies/utils.py:262: in get_dependant
    type_annotation, depends, param_field = analyze_param(
.venv/lib64/python3.11/site-packages/fastapi/dependencies/utils.py:443: in analyze_param
    field = create_response_field(
.venv/lib64/python3.11/site-packages/fastapi/utils.py:101: in create_response_field
    raise fastapi.exceptions.FastAPIError(
E   fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that typing.Optional[starlette.requests.Request] is a valid Pydantic field type. If you are using a return type annotation that is not a valid Pydantic field (e.g. Union[Response, dict, None]) you can disable generating the response model from the type annotation with the path operation decorator parameter response_model=None. Read more: https://fastapi.tiangolo.com/tutorial/response-model/

### Operating System

Linux

### Operating System Details

Fedora Linux 38

### FastAPI Version

0.105.0

### Pydantic Version

2.5.3

### Python Version

3.11.6

**Comment by Ventura94** [comment]
@jramiromolina Could you show me your ContentResponse object? The more complete the example, the easier it is to simulate the error and solve the problem.

**Reply by jramiromolina** [reply]
This is the object

```
class ContentResponse(BaseModel):
    """This the serializer exposed on the API"""

    id: int
    title: str
    slug: str
    text: str
    published: bool
    created_time: str
    tags: List[str]
    user_id: int

    def __init__(self, *args, **kwargs):
        # tags to model representation
        tags = kwargs.pop("tags", None)
        if tags and isinstance(tags, str):
            kwargs["tags"] = tags.split(",")
        super().__init__(*args, **kwargs)
```

and this is the "Content" class just in case is relevant

```
class Content(SQLModel, table=True):
    """This is an example model for your application.

    Replace with the *things* you do in your application.
    """

    id: Optional[int] = Field(default=None, primary_key=True)
    title: str
    slug: str = Field(default=None)
    text: str
    published: bool = False
    created_time: str = Field(
        default_factory=lambda: datetime.now().isoformat()
    )
    tags: str = Field(default="")
    user_id: Optional[int] = Field(foreign_key="user.id")

    # It populates a `.contents` attribute to the `User` model.
    user: Optional["User"] = Relationship(back_populates="contents")
```

**Reply by Ventura94** [reply]
I couldn't replicate the error on my premises, but you can try this way of declaring your ContentResponse:
````python
class ContentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    title: str
    slug: str
    text: str
    published: bool
    created_time: str
    tags: List[str] | str | None
    user_id: int

    @field_validator("tags", mode="before")
    def set_tags(cls, v, values):
        if isinstance(v, str):
            return v.split(",")
````
I hope it helps you, if not it would be good if you sent me the new error

**Reply by Ventura94** [reply, marked as answer]
I managed to replicate your error, the problem is that you are importing the wrong Request object somewhere in your code using ````python typing.Optional[starlette.requests.Request] ```` as a python annotation would be good that you import your Request dependency directly from FastAPI ````python
from fastapi import Request ```` and in the dependency you directly use ````python
request: Request ```` and not ````python request: typing.Optional[starlette.requests.Request] ````
I hope that helps

**Reply by jramiromolina** [reply]
Thanks a lot , I will give it a try
